server/policies/document.ts

```diff
@@ -94,8 +86,7 @@ allow(User, "update", Document, (user, document) => {
   if (document.deletedAt) {
     return false;
   }
-
-  if (cannot(user, "update", document.collection)) {
+  if (document.collection && cannot(user, "update", document.collection)) {
```

Comment:
Note that this `document.collection` acts more like a _skip_ rather than a _guard_. I'm not sure if there could be a case where `cannot(user, "update")` is true for a given user against _all collections_? Thoughts @tommoor ?

Reply:
There are several places where we had an `invariant` that would throw in the case of collection not existing, this was mainly to protect against mistakes in development where `collection` is not queried with the document but could protect against other cases too.

I think we want to change these to check `if (document.collectionId)` and then within that we can still use invariant if `document.collection` does not exist